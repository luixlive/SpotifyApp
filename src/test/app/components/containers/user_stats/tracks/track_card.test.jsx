import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import {
  trackCard as componentProps,
} from './../../../../../test_utils/components_props';
import initialState from './../../../../../test_utils/initial_state';
import TrackCard, {
  formatArtists,
  formatSongTime,
  PureTrackCard,
/* eslint-disable-next-line max-len */
} from './../../../../../../app/components/containers/user_stats/tracks/track_card';

describe('App Components - Track Card', () => {
  describe('Helpers', () => {
    it('formats various artists with commas', () => {
      const artists = [{
        id: '1',
        spotifyUrl: 'url1',
        name: 'name1',
      }, {
        id: '2',
        spotifyUrl: 'url2',
        name: 'name2',
      }];
      const anchorTagsArtistsList = formatArtists(artists);

      let wrapper = shallow(anchorTagsArtistsList[0]);
      expect(wrapper.props().href).toEqual(artists[0].spotifyUrl);
      expect(wrapper.text()).toEqual(`${artists[0].name}, `);

      wrapper = shallow(anchorTagsArtistsList[1]);
      expect(wrapper.props().href).toEqual(artists[1].spotifyUrl);
      expect(wrapper.text()).toEqual(`${artists[1].name}`);
    });

    it('formats ms into m:ss format', () => {
      expect(formatSongTime(120000)).toEqual('2:00');
      expect(formatSongTime(3000)).toEqual('0:03');
      expect(formatSongTime(1580000)).toEqual('26:20');
    });
  });

  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders no mobile', () => {
      const rendered = renderer.create(<PureTrackCard {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders mobile', () => {
      props.deviceMobile = true;
      const rendered = renderer.create(<PureTrackCard {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    let store;
    let wrapper;
    beforeAll(() => {
      store = mockStore(initialState);
      wrapper = mount((
        <Provider store={store}>
          <TrackCard place={0} />
        </Provider>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(PureTrackCard).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureTrackCard).props()))
        .toEqual(Object.keys(componentProps));
    });
  });
});
