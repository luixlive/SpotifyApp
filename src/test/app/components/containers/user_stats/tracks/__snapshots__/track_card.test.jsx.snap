// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - Track Card Snapshots renders mobile 1`] = `
<div>
  <h3
    className="ui left floated header"
  >
    0
  </h3>
  <div
    className="ui fluid card track"
    href={undefined}
    onClick={[Function]}
  >
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        className="ui image"
        src=""
      />
    </a>
    <div
      className="content"
    >
      <div
        className="header"
      >
        <h4
          className="ui header"
        >
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </h4>
      </div>
      <div
        className="meta"
      >
        <h5
          className="sub header"
        >
          0:00
        </h5>
      </div>
      <div
        className="description"
      >
        <h5
          className="sub header"
        >
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </h5>
      </div>
    </div>
    <div
      className="extra content"
    >
      Album: 
      <a
        href=""
      >
        
      </a>
      <br />
      Number: 0
    </div>
  </div>
</div>
`;

exports[`App Components - Track Card Snapshots renders no mobile 1`] = `
<div>
  <h2
    className="ui left floated header"
  >
    0
  </h2>
  <div
    className="ui fluid card track"
    href={undefined}
    onClick={[Function]}
  >
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        className="ui image"
        src=""
      />
    </a>
    <div
      className="content"
    >
      <div
        className="header"
      >
        <h3
          className="ui header"
        >
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </h3>
      </div>
      <div
        className="meta"
      >
        <h4
          className="sub header"
        >
          0:00
        </h4>
      </div>
      <div
        className="description"
      >
        <h4
          className="sub header"
        >
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            
          </a>
        </h4>
      </div>
    </div>
    <div
      className="extra content"
    >
      Album: 
      <a
        href=""
      >
        
      </a>
      <br />
      Number: 0
    </div>
  </div>
</div>
`;
